### ISSUE TITLE
file-loader not recognizing png on server restart

### ISSUE THREAD

**Issue by Always-Sunny** [webpack-contrib/file-loader#210]
node v8.5.0
"file-loader": "^0.11.2",

I'm getting the error of 'unexpected character' when trying to render a png (also failed on jpg). 
I have file-loader and url-loader, and both haven't worked on server start. But after the server is already running, if I then add the two lines of code (1) importing relative photo path (2) add <img> tag, then it'll load the photo... but once the server refreshes or I restart it, it errors out as an unexpected character

**This is my webpack config**

```
//webpack.config.js
import webpack from 'webpack';
import path from 'path';

module.exports = {
  entry: [
    'webpack-hot-middleware/client?reload=true', //note that it reloads the page if hot module reloading fails.
    'react-hot-loader/patch', // RHL patch
    './src/index.js'
  ],
  output: {
    path: __dirname + '/public',
    publicPath: "/",
    filename: 'bundle.js'
  },
  devtool: 'inline-source-map',
  plugins: [
    new webpack.HotModuleReplacementPlugin()
  ],
  module: {
   rules: [
     {test: /\.jsx?$/, exclude: /node_modules/, loaders: ['babel-loader']},

     {
       test: /\.scss$/,
       use: [
         { loader: "style-loader" },// creates style nodes from JS strings
         { loader: "css-loader" }, // translates CSS into CommonJS
         { loader: "sass-loader"} // compiles Sass to CSS
       ]
     },
     {
       test: /\.(png|jpg|gif)$/,
       use: [
         {
           loader: 'file-loader',
           options: {}
         }
       ]
     }
   ]
  }
};
```

**This is my header component in which I'm rendering the image**
```
//header.js
import React from 'react';
import { NavLink } from 'react-router-dom';
import headerLogo from '../../public/assets/images/ico.jpg';

const Header = () => (
  <div className="Header">
    <div className="Header-Container">

      <div className="header-title">
        <img className="header-logo-img" src={headerLogo} alt={'alt'}/>
        <h1><NavLink to="/" className="header-title-navlink header-title-text">TheDailyVibe</NavLink></h1>
      </div>

...
```

**This is the error**

> /Users/ryan/Desktop/Df/Coding/the-daily-vibe/node_modules/babel-core/lib/transformation/file/index.js:590
>       throw err;
>       ^
> 
> SyntaxError: /Users/ryan/Desktop/Df/Coding/the-daily-vibe/public/assets/images/ico.jpg: Unexpected character '�' (1:0)
> > 1 | �PNG
>     | ^
>   2 | 
>   3 | 

I've tried url-loader, image-webpack-loader, and file-loader, taking the code from the docs mostly @ webpack.js.org

GitHub repo: https://github.com/Always-Sunny/the-daily-vibe

To reiterate: I get an error, saying that the png is an unexpected character only when I try to start the server with the 'import img' and 'img tag' code. If I delete these though, the server runs, and then I add them, the image loads and there is no error.

**Comment by d3viant0ne** [comment]
@Always-Sunny - Update to `"file-loader": "^1.1.4"`

![screen shot 2017-09-30 at 7 13 07 pm](https://user-images.githubusercontent.com/8420490/31050548-85493496-a613-11e7-8eb2-236f24c1f88c.png)

**Comment by Panoplos** [comment]
Updating also fixed an error I was having with loading fonts. Some weird error like "Cannot find './options.json'" in a font file.

**Comment by Always-Sunny** [comment]
@d3viant0ne Didn't work for me.

Now @  
>  "file-loader": "^1.1.4",

**Comment by d3viant0ne** [comment]
Alright, I know it works on OSX on NodeJS 8.4 as you can see above.

What are you working with

**Comment by Always-Sunny** [comment]
macOS Sierra
v10.12.6

NodeJS 8.5.0

**Comment by Always-Sunny** [comment]
I created a bare minimum app to test file-loader (to check if maybe it was some config problem in my app). Using Node -v 8.6.0 and Fileloader 1.1.4, it's not working with even the most simple react/node/webpack/express config. Let me know, when you have some free time, if this repo works on your comp. Thank you.

https://github.com/Always-Sunny/bare-minimum/tree/fileloader

**Comment by d3viant0ne** [comment]
I've actually got some time right now. I'll take a look

**Comment by Always-Sunny** [comment]
Hey man, I figured out what's causing my error. It's when I server-render:

If I change:
```
//server.js with server render
const server = express();
import React from 'react';
import ReactDOMServer from 'react-dom/server';
import App from './src/App';

server.get('/', (req, res) => {
  res.render('index', {
    content: ReactDOMServer.renderToString(<App/)
  });
});
```

to this: 
```
//server.js without server render
const server = express();

server.get('/', (req, res) => {
  res.render('index', {
    content: "random text"
  });
});
```

I receive no error when I use relative imports in my <App/> component. However, once I even add:
`import App from './src/App';`
in my server.js file, the error throws. I can't import any react component in my server file without node throwing an error on the relative imports in my client components. 

I don't even have to use this line without the error throwing:
`ReactDOMServer.renderToString(<App/)`

As long as I import a react component into the server:
`import App from './src/App';`
Node won't accept my relative imports

What I can't figure out is how you got my own code to work. 
I've been experimenting with my bare-minimum repo to test all of this, and when I tried commenting out all of the react imports in my main repo (the one you cloned), it worked and I was able to render relative paths
